### PR TITLE
Fix checkstyle plugin 

### LINKED ISSUE #47: Ensure DCO Workflow Check
Coming from https://github.com/opensearch-project/project-meta/issues/17

A Developer Certificate of Origin is required on commits in the OpenSearch-Project.

See [doc.yml](https://github.com/opensearch-project/.github/blob/main/workflow/dco.yml) for an example workflow.  Ensure CONTRIBUTING.md to has a section on the DCO per the [project template](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#developer-certificate-of-origin). 

- [ ] DCO Check Workflow
- [ ] CONTRIBUTING.md DCO Section

### PR DESCRIPTION
**Pull request by eirsep**
### Description
Fix checkstyle plugin usage. 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,12 +54,6 @@ apply plugin: 'opensearch.opensearchplugin'
 apply plugin: 'opensearch.testclusters'
 apply plugin: 'opensearch.rest-test'
 
-
-checkstyle {
-    toolVersion = '8.24'
-    configFile file("checkstyle/checkstyle.xml")
```
Comment (Bukhtawar) [inline build.gradle:60]: why did we hv to remove this?

Comment (eirsep) [reply build.gradle:60]: will fix checkstyle  and raise a new revision.

**Review by peternied — COMMENTED** [review]
I see that the DCO check worked on this change which I would approve.  Please separated it into its own PR from the check style modifications

**Comment by eirsep** [comment]
> I see that the DCO check worked on this change which I would approve. Please separated it into its own PR from the check style modifications

Sure @peternied 
I didn't intend to fix checkstyle in this PR but the CI was failing due to it. Will raise 2 separate PRs.

File: build.gradle [hunk 2]
```diff
@@ -53,12 +53,14 @@ apply plugin: 'idea'
...
 apply plugin: 'opensearch.rest-test'
-
+apply plugin: 'checkstyle'
 
 checkstyle {
     toolVersion = '8.24'
     configFile file("checkstyle/checkstyle.xml")
 }
+checkstyleMain.ignoreFailures = true
+checkstyleTest.ignoreFailures = true
```
Comment (Bukhtawar) [inline build.gradle:63]: Do we plan to enable checkstyles later?

Comment (eirsep) [reply build.gradle:63]: Yes.  We will enable checkstyleMain and checkstyleTest tasks later.

**Comment by CEHENKLE** [comment]
Hey -- it looks like you're actively working on this, but I'm confused.  https://github.com/opensearch-project/asynchronous-search/pull/55 was raised, but closed.  Are you raising a separate PR to close #47 ?  

Thanks!
/C

**Comment by eirsep** [comment]
> Hey -- it looks like you're actively working on this, but I'm confused. #55 was raised, but closed. Are you raising a separate PR to close #47 ?
> 
> Thanks! /C

@CEHENKLE I have been suggested to raise a separate PR for DCO check workflow. I can raise that once this fix is merged.

File: build.gradle (hunk 2 above)
Comment (owaiskazi19) [reply build.gradle:63]: Opensearch Core has removed CheckStyle and are using Spotless as the formatting plugin from now. Ref: https://github.com/opensearch-project/OpenSearch/issues/1362

**Comment by eirsep** [comment]
@owaiskazi19 Have added a backlog item to use spotless formatting plugin

https://github.com/opensearch-project/asynchronous-search/issues/58

File: checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -6,9 +6,9 @@
 <module name="Checker">
     <property name="charset" value="UTF-8" />
 
-    <module name="SuppressionFilter">
-        <property name="file" value="${config_loc}/checkstyle_suppressions.xml" />
-    </module>
+<!--    <module name="SuppressionFilter">-->
```
Comment (owaiskazi19) [inline checkstyle/checkstyle.xml:9]: Can you add a link to a backlog issue to pick this up later?

**Review by CEHENKLE — APPROVED** [review]

**Comment by eirsep** [comment]
@CEHENKLE @owaiskazi19 
closing this CR and have raised #60 to add checkstyle plugin as a dependency .

**Pull request closed without merging**